near-sdk/src/test_utils/context.rs

```diff
@@ -56,13 +65,13 @@ impl VMContextBuilder {
         self
     }
 
-    pub fn epoch_height(&mut self, epoch_height: EpochHeight) -> &mut Self {
-        self.context.epoch_height = epoch_height;
+    pub fn block_timestamp(mut self, block_timestamp: u64) -> Self {
```

Comment:
Is it intentional that this method does not follow the rest of the methods API? (`self -> Self` instead of `&mut self -> &mut Self`)

Reply:
I changed all of them to Self in the other PR #293 
I don't really understand why `&mut Self` is used, as that really makes that object not very usable.

Reply:
`&mut Self` eliminates unnecessary copies and simplifies the conditional parameters:

```rust
let mut builder = Builder::new();
if something {
  builder.block_height(10);
}
if something_else {
  builder.epoch_height(10);
}
```

with `Self`:

```rust
let mut builder = Builder::new();
if something {
  builder = builder.block_height(10);
}
if something_else {
  builder = builder.epoch_height(10);
}
```

You can read more about cons and pros: http://web.archive.org/web/20161030171809/https://aturon.github.io/ownership/builders.html (`&mut Self` was recommended)